homeassistant/components/google_assistant/trait.py

```diff
@@ -1241,13 +1280,19 @@ def _generate(name, settings):
                 )
         elif self.state.domain == input_select.DOMAIN:
             modes.append(_generate("option", attrs[input_select.ATTR_OPTIONS]))
+        elif self.state.domain == humidifier.DOMAIN:
+            if humidifier.ATTR_AVAILABLE_MODES in attrs:
+                modes.append(_generate("mode", attrs[humidifier.ATTR_AVAILABLE_MODES]))
 
         payload = {"availableModes": modes}
 
         return payload
 
     def query_attributes(self):
         """Return current modes."""
+        if self.state.state == STATE_UNAVAILABLE:
+            return {"online": False}
+
```

Comment:
This is not necessary. `GoogleEnetty.query_serialize()` already takes care of this.
```suggestion
```

Reply:
Done